source/Unittests/TheService.cs

```diff
@@ -99,6 +99,23 @@ public void DoesNotThrowsExceptionForUnknownLocales(string locale)
             var dontCare = service.GetString(IdSrvConstants.Messages, MessageIds.MissingClientId);
         }
 
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public void GetLocaleReturnsDefaultWhenLocaleProviderReturnsNullOrWhitespace(string locale)
```

Comment:
Could you a unit test that uses the public method `GetString(string, string)` and verifies that a string is returned, even when the func is setup to returns nulls?